src/main/java/com/bobocode/svydovets/beans/scanner/BeanPostprocessorScanner.java

```diff
@@ -0,0 +1,42 @@
+package com.bobocode.svydovets.beans.scanner;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.reflections.Reflections;
+
+import com.bobocode.svydovets.beans.bpp.BeanPostProcessor;
+import com.bobocode.svydovets.exception.BeanPostprocessorInstantiationException;
+
+public class BeanPostprocessorScanner {
+
+    /**
+     * This method scan package to find classes implemented
+     * {@link BeanPostProcessor} and returns set of bean postprocessors
+     * created from these classes.
+     *
+     * @param packageName - package name that will be scanned
+     * @return - map of bean definition
+     */
+    public Set<BeanPostProcessor> scan(String packageName) {
+        Objects.requireNonNull(packageName, "The packageName cannot be null! Please specify the packageName.");
+        if (StringUtils.isEmpty(packageName)) {
```

Comment:
As an option we can remove 25 lane and check for null or empty using StringUtils, change the message for it if you want to apply